openmicroscopy/openmicroscopy
# Get thumbnails

# What this PR does

Adds 3 commits on top of #5158 to use getThumbnail() to load thumbnails one at a time when refreshing thumbnails after Save / Save to All and for Shares. This should mean that we don't get invalid thumbnails when trying to retrieve a SET while thumbnails are being recreated.

To test:

- This should fix the broken thumbnails after Save To All, most recently reported at https://github.com/openmicroscopy/openmicroscopy/pull/5158#issuecomment-293844986
- Also fixes thumbnails in shares (with 1 or more images) as reported at https://github.com/openmicroscopy/openmicroscopy/pull/5207#issuecomment-295641248

cc @aleksandra-tarkowska
